5_Stable_implementation/flip.cpp

```diff
@@ -0,0 +1,124 @@
+
+#include <torch/torch.h>
+
+struct Indexer {
+  Indexer(int64_t num_indexers, char** indexers, const int64_t* indexer_strides)
+    : num_indexers(num_indexers)
+    , indexers(indexers)
+    , indexer_strides(indexer_strides) {
+
+  }
+
+  int64_t num_indexers;
+  char** indexers;
+  const int64_t* indexer_strides;
+
+  int64_t get(int64_t idx) {
+    int64_t offset = 0;
+    for (int j = 0; j < num_indexers; j++) {
+      int64_t value = *(int64_t*)&indexers[j][idx * indexer_strides[j]];
+      offset += value;
+    }
```

Comment:
Maybe, for another benchmark iteration you can compare this implementation with
```c++
    int64_t offset = *(int64_t*)&indexers[0][idx * indexer_strides[0]];
    for (int j = 1; j < num_indexers; j++) {
      offset += *(int64_t*)&indexers[j][idx * indexer_strides[j]];
    }
```
which one is faster...